File: example/src/main/java/com/example/job/ExampleJob.java
package com.example.job;

import com.zipe.quartz.job.QuartzJobFactory;
import com.zipe.util.time.DateTimeUtils;
import lombok.extern.slf4j.Slf4j;
import org.quartz.JobExecutionContext;

/**
 * @author Gary.Tsai
 */
@Slf4j
public class ExampleJob extends QuartzJobFactory {

    public ExampleJob() {
    }

    @Override
    protected void executeJob(JobExecutionContext jobExecutionContext) throws Exception {
        log.info("{}執行, 當前的時間:{}", this.getClass(), DateTimeUtils.getDateNow());
    }
}

File: example-kotlin/src/main/kotlin/com/example/job/ExampleJob.kt
package com.example.job

import com.example.service.ExampleService
import com.zipe.quartz.job.QuartzJobFactory
import com.zipe.util.log.logger
import com.zipe.util.time.DateTimeUtils
import org.quartz.JobExecutionContext
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class ExampleJob: QuartzJobFactory() {

    @Autowired
    lateinit var exampleServiceImpl: ExampleService

    override fun executeJob(jobExecutionContext: JobExecutionContext?) {
        logger().info("Job class: ${this::class.simpleName}, 當前執行時間:${DateTimeUtils.dateTime}")
        val user = exampleServiceImpl.findUserByName("Tom")
        logger().info("取得使用者:${user.toString()}")
    }

}
